rewrite-javascript/build.gradle.kts

```diff
@@ -157,54 +157,42 @@ val npmInstallTemp = tasks.register<NpmTask>("npmInstallTemp") {
     val tempInstallDir = layout.buildDirectory.file("tmp/npmInstall")
     inputs.files(npmPack)
         .withPathSensitivity(PathSensitivity.RELATIVE)
+    outputs.dir(tempInstallDir)
     dependsOn(npmInitTemp)
 
     // Use a provider to defer evaluation until execution time
     args = provider { listOf("install", npmPack.get().archiveFile.get().asFile.absolutePath, "--omit=dev") }
     workingDir.set(tempInstallDir)
 }
 
-sourceSets {
-    main {
-        resources {
-            srcDir("src/main/generated-resources")
-        }
-    }
-}
-
-afterEvaluate {
-    tasks.named("licenseMain") {
-        dependsOn(createProductionPackage)
-    }
-}
-
-tasks.named<Jar>("sourcesJar") {
-    dependsOn("createProductionPackage")
-    exclude("production-package.zip")
-}
-
-tasks.named("processResources") {
-    dependsOn(createProductionPackage)
-}
-
 // Creates a production-ready package; writing it to `src/main/generated-resources` so that it will be included by IDEA
 val createProductionPackage by tasks.register<Zip>("createProductionPackage") {
     // Configure the tar output
     archiveFileName.set("production-package.zip")
-    destinationDirectory.set(layout.projectDirectory.dir("src/main/generated-resources"))
+    destinationDirectory.set(layout.buildDirectory.dir("generated/resources/distributions/"))
 
-    dependsOn(npmInstallTemp)
-
-    // Reference the actual directory where npm packages are installed
-    from(layout.buildDirectory.dir("tmp/npmInstall")) {
+    from(npmInstallTemp) {
         // Optionally exclude unnecessary files like package-lock.json, etc.
         // exclude("package-lock.json", ".npmrc")
     }
 }
 
 // Include production-ready package into jar
-tasks.named<Jar>("jar") {
+sourceSets {
+    main {
+        resources {
+            srcDir(createProductionPackage.destinationDirectory)
+        }
+    }
+}
+
+tasks.named<Jar>("sourcesJar") {
     from(createProductionPackage)
+    exclude("production-package.zip")
+}
+
+tasks.named<ProcessResources>("processResources") {
+    dependsOn(createProductionPackage)
 }
```

Comment:
callout: All of this is really a hack to make the IDE work correctly. The `production-package.zip` is a generated resource that would normally be included into the binary jar itself. Unfortunately, since IDEA doesn't generate the jar, but rather uses the raw classes and resources, we're stuck with forcing the `production-package.zip` to be generated earlier in the process than would typically be necessary, including it as a main resource for IDE support, and then having to filter it out of other tasks for which we don't want the generated binary to be included for.

Overall, not ideal by any means.